model-optimizer/mo/ops/slice.py

```diff
@@ -142,8 +142,6 @@ def infer(node: Node):
             slice_idx[axes[i]] = slice(starts[i], ends[i], steps[i])
         if input_value is None or any(is_dynamic_slice(s) for s in slice_idx):
             output_shape = get_shape_from_slice(input_shape, slice_idx)
-            if np.ma.any(output_shape <= 0):
-                raise Error('Output shape: {} of node "{}" contains non-positive values'.format(output_shape, node.name))
```

Comment:
(1) `get_shape_from_slice` newer returns shapes with negative values, (2) zero dimension values are correct and allowed by onnx & numpy

Reply:
Maybe 'never', not 'newer'?

Reply:
And more, allowing zero elements in output shapes of `Slice` can affect to all operations shapes infer. Are zero elements in input shapes correct for other operations, e.g., `Convolution`, `StridedSlice`, `Interpolate`, `NonMaxSuppression`, `FFT`, `DFFT`, etc?

Reply:
But what about other operations, e.g. `NonMaxSuppression`, `FFT`, `DFFT`, `Interpolate`, etc.?

Reply:
And more, are zero elements in shapes are correct for nGraph and plugins?

Reply:
> And more, are zero elements in shapes correct for nGraph and plugins?

yes, at least nGraph support such tensors. And usually tensors with 0 dimension sizes are concatenated with another tensors with non zero dimension. E.g.
```
tensor_1.shape = [1 3 225 225] 
tensor_2.shape = [1 0 255 255], 
res = Concat(tensor_1, tensor_2, axis=1), res.shape = [1 3 225 225].
```
about `FFT`, `NMS`, etc. I don't think they can (and must) return or accept tensors with zero size dimensions but Slice for sure must allow to return empty tensors with zero size dimensions.

Reply:
What about StridedSlice? Should we support 0-dimension tensors there?

Reply:
currently SS supports 0-dimension tensors.